src/middleware.js

```diff
@@ -56,26 +57,26 @@ export default function wrapper(context) {
         return;
       }
 
-      if (!res.get('Content-Type')) {
+      if (!res.getHeader('Content-Type')) {
```

Comment:
I think we can create `compatibility.js` file and put all helpers in one file with comments:
```
// `res.get` used by express
// `res.getHeader` used by fastify
// etc
```

Reply:
@evilebottnawi I think it'll be better to have getHeader, setHeader & end as the default functions as they're the default for Node.js http.ServerResponse and write `handleCompatibility.js`: 
```js
export default function handleCompatibility(ctx,res) {
  if(typeof res.getHeader === 'undefined'){
    res.getHeader = res.get
  }
  if(typeof res.setHeader === 'undefined'){
    res.setHeader = res.set
  }
  // ctx is koa context
  if(typeof res.end === 'undefined' && ctx){
    res.end = (content) => ctx.body = content;
  }
  res.setStatus = (code) => {
    res.statusCode = code;
    res.status = code;
  }
}
```
Note: Ctx is for koa js.

Reply:
Modified `res` and `req` is very unsafe, we should avoid it

Reply:
@evilebottnawi for koa I have to modify them because koa passes ctx and next for ex:
```js

app.use(async (ctx, next) => {
  await next();
  const rt = ctx.response.get('X-Response-Time');
  console.log(`${ctx.method} ${ctx.url} - ${rt}`);
});
```
What I did was to do this:
```js
  return async function middleware(...args) {
    let ctx;
    let req;
    let res;
    let next;
    // for detecting koa framework
    if(typeof args[1] === "function"){
      ctx = args[0];
      next = args[1]
      req = ctx.request
      res = ctx.response
    } else {
      req = args[0]
      res = args[1]
      next = args[2]
    }
   ......
```

Reply:
Let's focus on `fastify`, we will improve koa supporting in separate PR

I suggest to create `compatibility.js` as written above. For koa we will create wrapper around function to achieve this.

Here let's create `compatibility.js` file with two methonds:
```
function getResponseHeader() { }
function sendResponse() { }
```